### PR TITLE
OCPBUGS-29840: actuator: Fix DummyActuator to respect Manual mode upgrade gate

### PR DESCRIPTION
**Pull request by jstuever**
On unsupported platforms, the DummyActuator now checks the Manual mode upgrade annotation requirement. Previously, DummyActuator always reported upgradeable=true, bypassing the administrator confirmation annotation that Manual mode requires for upgrades.

This adds a Client field to DummyActuator and updates the Upgradeable method to delegate to UpgradeableCheck in Manual mode, ensuring consistent upgrade gating behavior across all platforms.

Assisted-by: Claude Sonnet 4.6

## Summary by CodeRabbit

* **Refactor**
  * Improved internal handling of unsupported platform scenarios through enhanced Kubernetes client integration
  * Refined upgrade checking logic for edge case operational modes

* **Tests**
  * Added comprehensive unit tests for upgrade behavior validation across different operational scenarios

### CODE REVIEW

File: pkg/operator/credentialsrequest/actuator/actuator.go [hunk 1]
```diff
@@ -74,16 +75,22 @@ func (a *DummyActuator) Delete(ctx context.Context, cr *minterv1.CredentialsRequ
...
 
 func (a *DummyActuator) Upgradeable(mode operatorv1.CloudCredentialsMode) *configv1.ClusterOperatorStatusCondition {
-	// Spec: Upgradeable=True means it is safe to upgrade. The DummyActuator
-	// always reports upgradeable since it performs no real cloud operations.
+	// In Manual mode, the admin must annotate the CCO config to confirm credentials
+	// are prepared for the next release. Delegate to UpgradeableCheck so unsupported
+	// platforms still respect that gate. rootSecret is unused in Manual mode.
+	if mode == operatorv1.CloudCredentialsModeManual && a.Client != nil {
+		return utils.UpgradeableCheck(a.Client, mode, types.NamespacedName{})
+	}
```
[the remark this answers is not in the document]
Comment (jstuever) [reply pkg/operator/credentialsrequest/actuator/actuator.go:91]: This is called in the dummy actuator, which is only created in one place. We always wire the client there, so the potential of this happening is very small. But, if it does happen for any reason, we want to error on the side of not blocking upgrades in this case.

**Comment by jstuever** [comment]
/override ci/prow/security
Unrelated, covered by other bugs
